packages/ckeditor5-emoji/tests/emojirepository.js

```diff
@@ -140,29 +142,61 @@ describe( 'EmojiRepository', () => {
 			expect( hasGroup2 ).to.be.undefined;
 		} );
 
-		it( 'should filter out unsupported emojis from the fetched emoji database', async () => {
-			isEmojiSupportedStub.callsFake( item => item.annotation !== 'microscope' );
+		it( 'should filter out unsupported ZWJ emojis from the fetched emoji database', async () => {
+			// Neutral face is mocked to be the only non ZWJ icon.
+			hasZwjStub.callsFake( emoji => emoji !== '😐️' );
+
+			// Unamused face is the only supported ZWJ emoji.
+			isEmojiSupportedStub.callsFake( item => item.annotation === 'unamused face' );
 
 			const response = JSON.stringify( [
-				{ annotation: 'neutral face', group: 0 },
-				{ annotation: 'unamused face', group: 0 },
-				{ annotation: 'microscope', group: 7 }
+				{ emoji: '😐️', annotation: 'neutral face', group: 0, version: 16 },
+				{ emoji: '😒', annotation: 'unamused face', group: 0, version: 15 },
+				{ emoji: '🔬', annotation: 'microscope', group: 7, version: 15 }
+			] );
+
+			fetchStubResolve( new Response( response ) );
+
+			editor = await editorPromise;
+
+			const emojiRepositoryPlugin = editor.plugins.get( EmojiRepository );
+			const hasNeutralFaceEmoji = emojiRepositoryPlugin._database.find( item => item.annotation === 'neutral face' );
+			const hasMicroscopeEmoji = emojiRepositoryPlugin._database.find( item => item.annotation === 'microscope' );
+			const hasUnamusedEmoji = emojiRepositoryPlugin._database.find( item => item.annotation === 'unamused face' );
```

Comment:
I know that data in mock are faked, but could we use a real example of a ZWJ emoji, as well as for a non-supported emoji in v16?

Reply:
Unfortunately we cannot perform fully unmocked tests for ZWJ icons, because we always install latest emoji on CI, and all ZWJ icons are going to be supported.

More detailed tests regarding emoji support are now extracted to `EmojiUtils` test.